thomasgauvin: add examples to workers kv (#20655)
cloudflare/cloudflare-docs: src/content/docs/kv/examples/routing-with-workers-kv.mdx

---
type: example
summary: Store routing data in Workers KV to route requests across various web servers with Workers
tags:
  - KV
pcx_content_type: configuration
title: Route requests across various web servers
sidebar:
  order: 5
description: Example of how to use Workers KV to build a distributed application configuration store.
---

import { Render, PackageManagers, Tabs, TabItem } from "~/components";

> Using Workers KV to store routing data to route requests across various web servers with Workers is an ideal use case for Workers KV.
> Routing workloads can have high read volume and Workers KV's low-latency reads can help ensure that routing decisions are made quickly and efficiently.

Reviewer: ```suggestion
Using Workers KV to store routing data to route requests across various web servers with Workers is an ideal use case for Workers KV. Routing workloads can have high read volume, and Workers KV's low-latency reads can help ensure that routing decisions are made quickly and efficiently.
```

> Routing can be helpful to route requests coming into a single Cloudflare Worker application to different web servers based on the request's path, hostname, or other request attributes.

Reviewer: ```suggestion
Routing can be helpful to route requests coming into a single Cloudflare Worker application to different web servers based on the request's path, hostname, or other request attributes.

```

> In single-tenant applications, this can be used to route requests to various origin servers based on the business domain (ex: requests to `/admin` routed to administration server, `/store` routed to storefront server, `/api` routed to the API server).

Reviewer: ```suggestion
In single-tenant applications, this can be used to route requests to various origin servers based on the business domain (for example, requests to `/admin` routed to administration server, `/store` routed to storefront server, `/api` routed to the API server).
```

> In multi-tenant applications, requests can be routed to the tenant's respective origin resources (ex: requests to `tenantA.your-worker-hostname.com` routed to server for Tenant A, `tenantB.your-worker-hostname.com` routed to server for Tenant B).

Reviewer: ```suggestion

In multi-tenant applications, requests can be routed to the tenant's respective origin resources (for example, requests to `tenantA.your-worker-hostname.com` routed to server for Tenant A, `tenantB.your-worker-hostname.com` routed to server for Tenant B).
```

Routing can also be used to implement [A/B testing](/reference-architecture/diagrams/serverless/a-b-testing-using-workers/), canary deployments, or blue-green deployments for your own external applications.
If you are looking to implement canary or blue-green deployments of applications built fully on Cloudflare Workers, see [Workers gradual deployments](/workers/configuration/versions-and-deployments/gradual-deployments/).

> ## Routing requests with Workers KV

Reviewer: ```suggestion
## Route requests with Workers KV
```
